gamemodes/nzombies/gamemode/weapons/sv_weps.lua

```diff
@@ -354,28 +354,19 @@ end)
 
 --hook.Add("OnEntityCreated", "nz.Weps.OnEntityCreated", nz.Weps.Functions.OnWepCreated)
 function GetPriorityWeaponSlot(ply)
-	if ply:HasPerk("mulekick") then
-		for i = 1, 3 do
-			local exists = false
-			for k,v in pairs(ply:GetWeapons()) do
-				if !exists and v:GetNWInt("SwitchSlot") == i then
-					exists = true
-				end
-			end
-			if !exists then return i end
-		end
-	else
-		for i = 1, 2 do
-			local exists = false
-			for k,v in pairs(ply:GetWeapons()) do
-				if !exists and v:GetNWInt("SwitchSlot") == i then
-					exists = true
-				end
+	local lastused = ply.LastUsedSWEP or ply:GetActiveWeapon()
+	local ie = 2
+	if ply:HasPerk("mulekick") then ie = 3 end
```

Comment:
i just realised, i accidently copy pasted this also into it. It just snippes down the for loop with the if case. It isn't any better or so, just for me it was easier to follow the events this function is doing better.